src/AWSCredentials.jl

```diff
@@ -254,27 +254,62 @@ using IniFile
 dot_aws_credentials_file() = get(ENV, "AWS_CONFIG_FILE",
                                  joinpath(homedir(), ".aws", "credentials"))
 
+dot_aws_config_file() = get(ENV, "AWS_CONFIG_FILE",
+                                 joinpath(homedir(), ".aws", "config"))
+
 """
 Load Credentials from [AWS CLI ~/.aws/credentials file]
 (http://docs.aws.amazon.com/cli/latest/userguide/cli-config-files.html).
 """
-
 function dot_aws_credentials()
 
     @assert isfile(dot_aws_credentials_file())
+    @assert isfile(dot_aws_config_file())
 
-    ini = read(Inifile(), dot_aws_credentials_file())
-
+    config_ini = read(Inifile(), dot_aws_config_file())
+    creds_ini = read(Inifile(), dot_aws_credentials_file())
```

Comment:
One or the other of these files may not exist, so `read(Inifile(), ...` will throw `ERROR: SystemError: opening file food: No such file or directory`. Probably best to have an if `isfile` around the block of code that uses that file.

The [cli doc says that .credentials has precedence over .config](https://docs.aws.amazon.com/cli/latest/topic/config-vars.html#id1) for creds. So the logic should be something like:
 - if there is a .config file:
    - if there are credentials for `profile` in the config file load them
    - if there is a `role_arn` and `source_profile` for `profile, grab the `role_arn` and overwrite the `profile` variable.
    - try to load credentials for `profile` in the config file again in case the source_profile refers to another profile in the config file.
 - if there is a .credentials file:
    - if there are credentials for `profile` in the .credentials file load them (overwriting whatever was loaded from the .config file.
 - create the AWSCredentials object
 - If there is a `role_arn`,  call AssumeRole and create a new AWSCredentials object
 - return the  AWSCredentials object